staging/src/k8s.io/client-go/transport/websocket/client.go

```diff
@@ -61,42 +62,12 @@ type ConnectionHolder interface {
 	Connection() *gwebsocket.Conn
 }
 
-// RoundTripper knows how to establish a connection to a remote WebSocket endpoint and make it available for use.
-// RoundTripper must not be reused.
-type RoundTripper struct {
```

Comment:
This is where I think we are streching the limits, but is a consequence of multiple things, websockets require a request transaction to upgrade the connection, so you actuall rountrip, but you yield a connection, so is not really a dialer. Instead of Roundtripper I chosen to call it `Client` with a method `Connect` and it hides internally the `RoundTrip` + `Dial` functionality. Connect returns a gorilla websocket object, but we are already exposing it in other places and I do not think we have more alternatives since we are already coupled to this implementation on the exec implementation